### ISSUE TITLE
NoSuchBeanDefinitionException: No qualifying bean of type 'graphql.GraphQL' available: 

### ISSUE THREAD

**Issue by afuyo** [graphql-java-generator/graphql-maven-plugin-project#56]
Hi, 
I run into this error when running graphql-java-generator and graphq-java-kickstart. Is there a solution to that? Here is the snippet from my pom.xml 
```
<properties>
	<graphql-maven-plugin.version>1.11</graphql-maven-plugin.version>
</properties>
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.4.0</version>
		<relativePath/> <!-- lookup parent from repository -->
	</parent>
/---/
	<dependency>
			<groupId>com.graphql-java-kickstart</groupId>
			<artifactId>graphql-spring-boot-starter</artifactId>
			<version>8.0.0</version>
		</dependency>

		<dependency>
			<groupId>com.graphql-java-generator</groupId>
			<artifactId>graphql-java-runtime</artifactId>
			<version>${graphql-maven-plugin.version}</version>
		</dependency>
```

**Comment by etienne-sf** [comment]
Hum, 

  The graphql-java-generator plugin should be used stand-alone.

Why did you add graphq-java-kickstart to the classpath ?

Etienne

**Comment by afuyo** [comment]
Hi Etienne, I wanted to use java beans generated by graphql-java-generator in my graphql spring boot service. Basically following the same idea as in GraphqQL Forum Tutorial server. And I can see that you also use graphql-java-kickstart but version 6.0.1. I use version 8.0.0 and suppose that is  causing the problem? I kind of can make it work  but I sitll have some problems with generated method stubs. 
Graphql-java-genarator generated mutation will look like this: `public Post createPost(DataFetchingEnvironment dataFetchingEnvironment, PostInput post);`
But it now needs to look like this: `public Post createPost(PostInput post);`
I tried to generate these myself by using annotations but can't find the last piece, the input argument. Is there an easy way to retrieve that information? The graphql schema I will be using is large, so editing that manually would be really a tedious work. 

Artur

**Comment by etienne-sf** [comment]
Hello,


  From com.graphql-java-kickstart, the plugin only uses the graphiql tool, which is a very convenient interface to test the GraphQL interface.

  But the graphql-spring-boot-starter (not that there is no "i" in graphql here) is the java code. And it an not work together with the plugin generated code.  BTW, I'm wondering to embed the generated code into kickstart, to include this plugin into the kickstart project. But it's currently only an idea.


About the mutation generated code, why would like to loose the dataFetchingEnvironment ?
If you loose it, then you can have access to the context (directive, parameters...).

And if you really want that, you can change the Velocity templates that generates the code.
Take a look at this page : https://graphql-maven-plugin-project.graphql-java-generator.com/customtemplates.html

Etienne

**Comment by afuyo** [comment]
Hi,
I don't want to lose it but graphql-java-kickstart does. When using it I need to implement `graphql.kickstart.tools.GraphQLMutationResolver` and method call like this: `public Post createPost(PostInput post);`. 

I have no opinion on that:) My use case is that I want to use GraphQLschema on top of Kafka. So I will be mapping between GraphQL mutations and Kafka producers. 

Since the schema we use is quite large I'd like to auto generate this mappings. That is why I need return type, method and input parameter. I can sort of guess input parameter since they follow naming conventions but it's risky. Most of the time it will be 1 to 1 mapping and I use Mapstruct for that.

**Comment by afuyo** [comment]
Actually I've just learned that  graphql.kickstart.tools  use dataFetchingEnvironment but in slightly different order ` public Person createOnePerson(PersonCreateInput input,DataFetchingEnvironment dataFetchingEnvironment)`

**Comment by etienne-sf** [comment]
So, if the generated code contain a method _public Person createOnePerson(PersonCreateInput input,DataFetchingEnvironment dataFetchingEnvironment)_, this would solve your issue ?

This is not that trivial to do in the plugin, as if the generated code is changed, it would break the code from the other people that use this plugin.
So there really are three ways to deal with:
* You use the Custom Template feature, to adapt the generated code to your needs. You'll have to change this file: _graphql-maven-plugin-project\graphql-maven-plugin-logic\src\main\resources\templates\server_GraphQLDataFetchersDelegate.vm.java_
* The plugin change the parameter orders, but break existing code. I would not do that.
* A plugin parameter allows to choose the parameter order. I'll think around this one. But it would be with a larger point of views, that is: make this plugin works better with kickstart.

My recommandation: give a try to the custom template feature.

Etienne

**Comment by etienne-sf** [comment]
Hello,

The original issue is due to `graphql-java-runtime` containing some server dependencies.

This is solved in the 1.18.1, by switching to `graphql-java-client-runtime`
